src/app/routes/applications/applications.component.ts

```diff
@@ -138,11 +138,14 @@ export class ApplicationsComponent implements OnInit, AfterViewInit, OnDestroy {
     this.store.dispatch(OrganizationActions.createSubForParent());
   }
 
-  async ngOnInit() {
-    this.isIamEwcOwner = await this.iamService.domainsService.isOwner({
-      domain: this.envService.rootNamespace,
-    });
-    this.orgRequestButtonText = this.envService.production
+  ngOnInit() {
+    from(
+      this.iamService.domainsService.isOwner({
+        domain: this.envService.rootNamespace,
+      })
+    ).subscribe({ next: (isOwner) => (this.isIamEwcOwner = isOwner) });
```

Comment:
What this the fix? Is it that `ngOnInit` can't be async?

Reply:
Nah, i just slightly refactored it to make tests more readable. The problem was that UI was checking `production` property instead of `isIamEwcOwner`. Build on dev env is set with production flag as true, that's why it stopped working. This now code checks if email is defined or not. If not, then we are navigating to role enrolment.

Reply:
@dawidgil 
> This now code checks if email is defined or not.

where is this check done? Is this related to the `this.orgRequestButtonText = this.envService.orgRequestEmail` line two lines below this section?

Reply:
yes, exactly.